crates/blockifier/src/execution/contract_class.rs

```diff
@@ -546,13 +548,28 @@ impl SierraContractClassV1 {
         Some(contract_entrypoint_to_entrypoint_selector(entrypoint))
     }
 
-    pub fn try_from_json_string(
-        raw_contract_class: &str,
-    ) -> Result<SierraContractClassV1, ProgramError> {
+    pub fn try_from_json_string(raw_contract_class: &str) -> Result<Self, ProgramError> {
+        fn compile(sierra_program: &SierraProgram) -> AotNativeExecutor {
+            let native_context = cairo_native::context::NativeContext::new();
+            let native_program = native_context.compile(sierra_program, None).unwrap();
+            AotNativeExecutor::from_native_module(native_program, OptLevel::Default)
+        }
+
         let sierra_contract_class: SierraContractClass = serde_json::from_str(raw_contract_class)?;
-        let contract_class: SierraContractClassV1 = sierra_contract_class.try_into()?;
+        // todo(rodro): we are having two instances of a sierra program, one it's object form
+        // and another in its felt encoded form. This can be avoided by either:
+        //   1. Having access to the encoding/decoding functions
+        //   2. Refactoring the code on the Cairo mono-repo
 
-        Ok(contract_class)
+        let sierra_program = sierra_contract_class.extract_sierra_program().unwrap();
+        let executor = compile(&sierra_program);
+
+        Ok(Self(Arc::new(SierraContractClassV1Inner {
```

Comment:
We should re-name from `SierraContractClass...` definitions to `NativeContractClass`. This naming is more clear in my opinion